// Copyright 2018 The Chromium Authors. All rights reserved.
// Use of this source code is govered by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

syntax = "proto3";

package monorail;

import "api/api_proto/common.proto";
import "api/api_proto/features_objects.proto";

service Features {
  rpc ListHotlistsByUser (ListHotlistsByUserRequest) returns (ListHotlistsByUserResponse) {}
  rpc GetHotlistStarCount (GetHotlistStarCountRequest) returns (GetHotlistStarCountResponse) {}
  rpc StarHotlist (StarHotlistRequest) returns (StarHotlistResponse) {}
}


// Next available tag: 3
message ListHotlistsByUserRequest {
  RequestTrace trace = 1;
  UserRef user = 2;
}


// Next available tag: 2
message ListHotlistsByUserResponse {
  repeated Hotlist hotlists = 1;
}


// Next available tag: 3
message GetHotlistStarCountRequest {
  RequestTrace trace = 1;
  HotlistRef hotlist_ref = 2;
}


// Next available tag: 2
message GetHotlistStarCountResponse {
  uint32 star_count = 1;
}


// Next available tag: 4
message StarHotlistRequest {
  RequestTrace trace = 1;
  HotlistRef hotlist_ref = 2;
  bool starred = 3;
}


// Next available tag: 2
message StarHotlistResponse {
  uint32 star_count = 1;
}
